card/mod/05_standard/set/abstract/attachment.rb

```diff
@@ -54,7 +54,10 @@ def assign_attachment file, original_filename
   update_column(:db_content,attachment.db_content(mod: load_from_mod))
   expire
 end
-event :save_original_filename, after: :validate_name, when: proc {|c| !c.preliminary_upload? && !c.save_preliminary_upload? && c.attachment_changed?} do
+event :save_original_filename,
+      after: :validate_name,
+      when: proc {|c| c.attachment.file.present? && !c.preliminary_upload? &&
```

Comment:
Avoid using `{...}` for multi-line blocks.<br>Block body expression is on the same line as the block start.